src/main/java/org/primefaces/component/inplace/InplaceRenderer.java

```diff
@@ -193,4 +198,16 @@ public void encodeChildren(FacesContext context, UIComponent component) throws I
     public boolean getRendersChildren() {
         return true;
     }
+
+    private boolean hasRequiredChildren(Inplace inplace) {
+        List<UIComponent> children = inplace.getChildren();
+        for (int i = 0; i < children.size(); i++) {
+            UIComponent c = children.get(i);
+            if (c instanceof UIInput && ((UIInput) c).isRequired()) {
```

Comment:
I think it needs support of composite also. See https://github.com/primefaces/primefaces/blob/271a52100de06d8c8a339025e9a0b4c5500e31a0/src/main/java/org/primefaces/component/outputlabel/OutputLabelRenderer.java#L70

Reply:
I think you might be right, I have never seen that ContextCallback code before.  I would have to study it some more to understand what it's doing.